### PR TITLE
Azure secrets engine

### PR DESCRIPTION
**Pull request by devops-rob**
Adding resources to enable flexible configuration of the azure secrets engine in vault.

Adding License file

### CODE REVIEW

File: azure_variables.tf [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+variable "azure_client_secret" {
+  type        = string
+  default     = null
+  description = "Client secret for Azure Service Principal. WARNING - This will be written to the state file in plain text"
+}
+
+variable "azure_environment" {
+  type        = string
+  default     = "AzurePublicCloud"
+  description = "The Azure cloud environment. Valid values: AzurePublicCloud, AzureUSGovernmentCloud, AzureChinaCloud, AzureGermanCloud.  Defaults to AzurePublicCloud"
```
Comment (ksatirli) [inline azure_variables.tf:28]: You could omit this from the description as the `can` check will inform module users of a transgression and provide values

Comment (devops-rob) [reply azure_variables.tf:28]: good point

Comment (devops-rob) [reply azure_variables.tf]: just pushed the change @ksatirli

File: azure_variables.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+variable "azure_secret_backend_ttl" {
+  type        = number
+  default     = 3600
+  description = "Default TTL for Azure secret backend."
+}
+
+variable "use_resource_group" {
+  type        = bool
+  default     = false
+  description = "Toggle to enable usage of Resource Groups for Azure Role Scopes.  When set to true, resource_group_identifier must be set."
```
Comment (ksatirli) [inline azure_variables.tf:57]: double space

File: azure_variables.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+variable "azure_role" {
+  type        = string
+  default     = "Reader"
+  description = "Azure role to assigned to service principal. Must be one of Reader, Contributor, Owner"
```
Comment (ksatirli) [inline azure_variables.tf:63]: Consider adding a validation rather than outlining the values in the `description`

File: main.tf [hunk 4]
```diff
@@ -41,4 +39,33 @@ resource "vault_aws_secret_backend_role" "aws_backend_role" {
...
+  count = contains(var.secrets_engines, "azure") ? 1 : 0
+  path  = "azure"
+
+  subscription_id = var.azure_subscription_id
+  tenant_id       = var.azure_tenant_id
+
+  client_id     = var.azure_client_id
+  client_secret = var.azure_client_secret
+  environment   = var.azure_environment
+
```
Comment (ksatirli) [inline main.tf:54]: consider removing this empty line

File: examples/azure_defaults/main.tf [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+module "example_defaults" {
+  source          = "../../"
+  secrets_engines = ["azure"]
+
+  azure_subscription_id = "11111111-2222-3333-4444-44444444444"
+  azure_tenant_id       = "11111111-2222-3333-4444-44444444444"
+
+  use_resource_group             = false
+  azure_secret_backend_role_name = "test_role"
+
```
Comment (ksatirli) [inline examples/azure_defaults/main.tf:15]: consider removing this empty line

**Comment by ksatirli** [comment]
Overall, this looks great! I've got some nits that I think are worth cleaning up, but otherwise this is good to go.

File: examples/azure_defaults/main.tf (hunk 5 above)
Comment (devops-rob) [reply examples/azure_defaults/main.tf]: new commit with changes suggested :-D

**Pull request merged**